### PR TITLE
Label e2e runner, operator and stack pods for log correlation

### LINKED ISSUE #2606: Instrument tests for E2E test monitoring
To allow correlating Stack, Operator and test runner logs we should annotate all resources with at least:
- provider 
- test run input (stack/k8s version)
- test name 

and inject those to all E2E test logs.

### PR DESCRIPTION
**Pull request by david-kow**
This PRs enables:
1. labeling operator pod, e2e test runner pod and stack pods with test name that is currently being ran. Note that while labels are added to stack pods on creation, e2e test runner and operator pods are relabelled during each test.
2. injecting additional correlation information (pipeline name, build number, infrastructure provider, k8s cluster name, k8s version, stack version) in each log.

These will allow to correlate each log with particular test in particular test run.

Note:
- Today, we use a raw autodiscovery annotation on the operator which prevents from appending any other config via other annotations or filebeat config, so for non-local e2e test runs raw annotation is not added now. Instead, filebeat config [adds needed config](https://github.com/elastic/cloud-on-k8s/compare/master...david-kow:labels_for_e2e?expand=1#diff-0a7b2591f7d2ce1f79d60ff9592eac71R30).
- Pods had to have labels added in multiple places:
  - when creating a Builder (Kibana/APM)
  - when adding node sets to a Builder (ES)
  - when setting PodTemplate (ES)
  - when not using Builder constructor (Smoke/Samples tests)

  Right now, both main resource and pods are labelled. Labelling pods can be removed when https://github.com/elastic/cloud-on-k8s/issues/2652 is implemented.

Resolves #2606.

### CODE REVIEW

**Comment by david-kow** [comment]
jenkins test this please

File: test/e2e/test/utils.go [hunk 1]
```diff
@@ -118,3 +119,57 @@ func ValidateBuilderHashAnnotation(pod corev1.Pod, hash string) error {
...
+func LabelTestPods(c k8s.Client, ctx Context, key, value string) error {
+	// label operator pod
+	if err := labelPod(
+		c,
+		ctx.Operator.Name+"-0",
+		ctx.Operator.Namespace,
+		key,
+		value); err != nil {
+		return err
+	}
```
Comment (pebrc) [inline test/e2e/test/utils.go]: What if the operator is running locally as well via `make run`?

File: .ci/setenvconfig [hunk 2]
```diff
@@ -46,6 +46,12 @@ MONITORING_SECRETS = /go/src/github.com/elastic/cloud-on-k8s/.ci/monitoring-secr
 
 OPERATOR_IMAGE = $JKS_PARAM_OPERATOR_IMAGE
 TEST_TIMEOUT = 10m
+
+PIPELINE=e2e/aks
+BUILD_NUMBER=$BUILD_NUMBER
+E2E_PROVIDER=aks
```
[the remark this answers is not in the document]
Comment (david-kow) [reply .ci/setenvconfig]: `CLOUD_PROVIDER` sounded weird to me for kind, do you think I should change it anyway?

File: Makefile [hunk 3]
```diff
@@ -379,6 +379,11 @@ e2e-run:
 		--log-verbosity=$(LOG_VERBOSITY) \
 		--log-to-file=$(E2E_JSON) \
 		--test-timeout=$(TEST_TIMEOUT) \
+		--pipeline=$(PIPELINE) \
+		--build-number=$(BUILD_NUMBER) \
+		--provider=$(E2E_PROVIDER) \
+		--clusterName=$(CLUSTER_NAME) \
+		--kubernetes-version=$(KUBERNETES_VERSION) \
```
[the remark this answers is not in the document]
Comment (david-kow) [reply Makefile]: I [agree](https://github.com/elastic/cloud-on-k8s/pull/1364#discussion_r308648630) :) I'll file an issue.

File: .ci/setenvconfig (hunk 2 above)
[the remark this answers is not in the document]
Comment (pebrc) [reply .ci/setenvconfig]: Right, I did not think about that case.

**Pull request merged**